manifests/params.pp

```diff
@@ -53,14 +53,14 @@
 
   $snmp_agentx_timeout = getvar('::snmp_agentx_timeout')
   if $snmp_agentx_timeout {
-    $agentx_timeout = $::snmp_agentx_timeout
+    $agentx_timeout = 0 + $::snmp_agentx_timeout
```

Comment:
The rest of the params file says ENC vars are often strings and uses `str2bool` a lot.

Reply:
Do we want to keep supporting top scoped (global) parameters? This could easily be solved in hiera?

Reply:
Given it's a major version bump anyway, it's a good time to get rid of them IMHO

Reply:
And Foreman can't set types for global parameters AFAIK., but can set them on class parameters in the ENC. Yet another reason to remove the global scoped parameters IMHO.

Reply:
Maybe merge this for now and remove all the global parameter cruft in a separate PR?

Reply:
That'd be https://github.com/voxpupuli/puppet-snmp/pull/145